halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/providers/AbstractProviderEnableDisableCommand.java

```diff
@@ -52,8 +53,8 @@ public String getDescription() {
 
   private void setEnable() {
     DaemonService service = Daemon.getService();
-    String currentDeployment = service.getCurrentDeployment();
-    service.setProviderEnabled(currentDeployment, getProviderName(), !noValidate, isEnable());
+    String currentDeployment = ResponseUnwrapper.get(service.getCurrentDeployment());
```

Comment:
Instead of having `ResponseUnwrapper` everywhere, have you considered returning a proxy class from `Daemon.getService` that will do the unwrapping for you?